tools/devops/automation/build-pipeline.yml

```diff
@@ -0,0 +1,193 @@
+  #    displayName: 'Device tests'
+  #    timeoutInMinutes: 1000
+  #
+  #    strategy:
+  #      matrix:
+  #        iOS32: # TODO: This bots should be moved to the ddfun pool
+  #          deviceDemands: 'xismoke-32'
+  #          testsLabels: '--label=run-ios-32-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+  #          poolName: 'VSEng-Xamarin-QA'
+  #        iOS64:
+  #          deviceDemands: 'ios'
+  #          testsLabels: '--label=run-ios-64-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+  #          poolName: 'VSEng-Xamarin-Mac-Devices'
+  #        tvOS:
+  #          deviceDemands: 'tvos'
+  #          testsLabels: '--label=run-tvos-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+  #          poolName: 'VSEng-Xamarin-Mac-Devices'
+  #
+  #    pool:
+  #      name: $(poolName)
+  #      demands: $(deviceDemands)
+  #      workspace:
+  #        clean: all
+  #
+  #    steps:
+  #    - template: templates/device-tests.yml
+  #
+  # Unfortunally, variable expansion will not happen on the right time, and will result in an agent error, to fix that
+  # we use a template for the test and we set each of the jobs. Not ideal, but is only a 3 jobs matrix
+
+- template: templates/devices/stage.yml
+  parameters:
+    devicePrefix: 'iOS32b'
+    execute: 'runDevice32b'
+    stageName: 'iOS32b Device Tests'
+    iOSDevicePool: 'VSEng-Xamarin-QA'
+    useXamarinStorage: False
+    testsLabels: '--label=run-ios-32-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+    statusContext: 'VSTS: device tests iOS32b'
+    iOSDeviceDemand: 'xismoke-32'
+    vsdropsPrefix: ${{ variables.vsdropsPrefix }}
+    keyringPass: $(xma-password)
+
+- template: templates/devices/stage.yml
+  parameters:
+    devicePrefix: 'iOS64'
+    execute: 'runDevice64b'
+    stageName: 'iOS64 Device Tests'
+    iOSDevicePool: 'VSEng-Xamarin-Mac-Devices'
+    useXamarinStorage: False
+    testsLabels: '--label=run-ios-64-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+    statusContext: 'VSTS: device tests iOS'
+    iOSDeviceDemand: 'ios'
+    vsdropsPrefix: ${{ variables.vsdropsPrefix }}
+    keyringPass: $(xma-password)
+
+- template: templates/devices/stage.yml
+  parameters:
+    devicePrefix: 'tvOS'
+    execute: 'runDeviceTv'
+    stageName: 'tvOS Device Tests'
+    iOSDevicePool: 'VSEng-Xamarin-Mac-Devices'
+    useXamarinStorage: False
+    testsLabels: '--label=run-tvos-tests,run-non-monotouch-tests,run-monotouch-tests,run-mscorlib-tests'
+    statusContext: 'VSTS: device tests tvOS'
+    iOSDeviceDemand: 'tvos'
+    vsdropsPrefix: ${{ variables.vsdropsPrefix }}
+    keyringPass: $(xma-password)
+
+- template: templates/mac-tests.yml
+  parameters:
+    stageName: 'Mac Mojave Tests'
+    macPool: 'Hosted Mac Internal Mojave'
+
+- template: templates/mac-tests.yml
+  parameters:
+    stageName: 'Mac High Sierra Tests'
+    macPool: 'Hosted Mac Internal'
```

Comment:
questions
* how do the pools handle Catalina and BigSur ?
* how do the pools handle 11.0 and 11.1 ? all updated asap vs different pools ?
* any pool for beta macOS, e.g. upcoming 11.2 ?

Reply:
ATM we have a pool for each, but I updated the agent to be able to filter using the Aget.OS, @mjbond-msft was going to create a pool with all the needed devices, in that case, we woud use a single pool for 11. and 11. Etc

@mjbond-msft can we have beta bots on those pools with unstable OS? I don't think it would be an issue as long as the agent works.

Reply:
We've already have a pool `VSEng-Xamarin-RedmondMacTextPool-iOS`, which currently has one Big Sur machine. We can add machines as needed to that pool